{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "export cwd=$(pwd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/cluster/gs_lab/common/co_chip\n"
     ]
    }
   ],
   "source": [
    "echo $cwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "export SRR_RUN_TABLE=$cwd/SraRunTable.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory `run': File exists\n"
     ]
    }
   ],
   "source": [
    "cd $cwd\n",
    "mkdir run\n",
    "cd run"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "rm -rf *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir index\n",
    "mkdir annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2019-08-16 11:00:12--  ftp://ftp.ensembl.org/pub/release-97/fasta/mus_musculus/dna//Mus_musculus.GRCm38.dna.toplevel.fa.gz\n",
      "           => 'Mus_musculus.GRCm38.dna.toplevel.fa.gz'\n",
      "Resolving ftp.ensembl.org... 193.62.193.8\n",
      "Connecting to ftp.ensembl.org|193.62.193.8|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /pub/release-97/fasta/mus_musculus/dna/ ... done.\n",
      "==> SIZE Mus_musculus.GRCm38.dna.toplevel.fa.gz ... 842816838\n",
      "==> PASV ... done.    ==> RETR Mus_musculus.GRCm38.dna.toplevel.fa.gz ... done.\n",
      "Length: 842816838 (804M) (unauthoritative)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "cd annotation\n",
    "wget ftp://ftp.ensembl.org/pub/release-97/fasta/mus_musculus/dna//Mus_musculus.GRCm38.dna.toplevel.fa.gz\n",
    "cd .."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd index\n",
    "bowtie2-build ../annotation/Mus_musculus.GRCm38.dna.toplevel.fa.gz mm9\n",
    "bowtie2-inspect -n mm9\n",
    "export INDEX=$(pwd)/mm9\n",
    "cd .."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "python3 $cwd/parse_sra_runtable.py $SRR_RUN_TABLE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "find ./*_*/SRX* -mindepth 0 -maxdepth 0 | parallel -j 17 \"$cwd/align.sh {} 2\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "find ./*_*/SRX*/merged -mindepth 0 -maxdepth 0 | parallel -j 40 $'samtools view -h {}/all.bam |awk \"sqrt((\\$9)^2) < 230\" |\\samtools view -bS - > {}/all-filtered.bam'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "find ./*_*/SRX*/merged -mindepth 0 -maxdepth 0 | parallel -j 40 \"samtools index {}/all-filtered.bam\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "BAM file : ./H3K27ac_H3K27ac/SRX1943296/merged/all-filtered.bam\n",
      "Sample size: 10401\n",
      "\n",
      "Fragment lengths:\n",
      "Min.: 0.0\n",
      "1st Qu.: 113.0\n",
      "Mean: 147.74896644553408\n",
      "Median: 151.0\n",
      "3rd Qu.: 187.0\n",
      "Max.: 229.0\n",
      "Std: 48.63484644531806\n",
      "MAD: 37.0\n",
      "Len. 10%: 83.0\n",
      "Len. 20%: 104.0\n",
      "Len. 30%: 122.0\n",
      "Len. 40%: 136.0\n",
      "Len. 60%: 166.0\n",
      "Len. 70%: 179.0\n",
      "Len. 80%: 194.0\n",
      "Len. 90%: 211.0\n",
      "Len. 99%: 228.0\n",
      "\n",
      "\n",
      "Read lengths:\n",
      "Sample size: 10401\n",
      "\n",
      "Min.: 38.0\n",
      "1st Qu.: 38.0\n",
      "Mean: 38.0\n",
      "Median: 38.0\n",
      "3rd Qu.: 38.0\n",
      "Max.: 38.0\n",
      "Std: 0.0\n",
      "MAD: 0.0\n",
      "Len. 10%: 38.0\n",
      "Len. 20%: 38.0\n",
      "Len. 30%: 38.0\n",
      "Len. 40%: 38.0\n",
      "Len. 60%: 38.0\n",
      "Len. 70%: 38.0\n",
      "Len. 80%: 38.0\n",
      "Len. 90%: 38.0\n",
      "Len. 99%: 38.0\n",
      "\n",
      "/homes/dhelekal/anaconda3/envs/experiment_env/lib/python3.5/site-packages/matplotlib/axes/_axes.py:6499: MatplotlibDeprecationWarning:\n",
      "\n",
      "\n",
      "The 'normed' kwarg was deprecated in Matplotlib 2.1 and will be removed in 3.1. Use 'density' instead.\n",
      "\n",
      "\n",
      "\n",
      "BAM file : ./H3K27ac_H3K4me3/SRX1943227/merged/all-filtered.bam\n",
      "Sample size: 7781\n",
      "\n",
      "Fragment lengths:\n",
      "Min.: 0.0\n",
      "1st Qu.: 110.0\n",
      "Mean: 145.32078139056676\n",
      "Median: 147.0\n",
      "3rd Qu.: 184.0\n",
      "Max.: 229.0\n",
      "Std: 49.841475121517014\n",
      "MAD: 37.0\n",
      "Len. 10%: 79.0\n",
      "Len. 20%: 101.0\n",
      "Len. 30%: 118.0\n",
      "Len. 40%: 133.0\n",
      "Len. 60%: 163.0\n",
      "Len. 70%: 177.0\n",
      "Len. 80%: 193.0\n",
      "Len. 90%: 211.0\n",
      "Len. 99%: 227.0\n",
      "\n",
      "\n",
      "Read lengths:\n",
      "Sample size: 7781\n",
      "\n",
      "Min.: 34.0\n",
      "1st Qu.: 42.0\n",
      "Mean: 41.862228505333505\n",
      "Median: 42.0\n",
      "3rd Qu.: 42.0\n",
      "Max.: 42.0\n",
      "Std: 1.0407646095968688\n",
      "MAD: 0.0\n",
      "Len. 10%: 42.0\n",
      "Len. 20%: 42.0\n",
      "Len. 30%: 42.0\n",
      "Len. 40%: 42.0\n",
      "Len. 60%: 42.0\n",
      "Len. 70%: 42.0\n",
      "Len. 80%: 42.0\n",
      "Len. 90%: 42.0\n",
      "Len. 99%: 42.0\n",
      "\n",
      "/homes/dhelekal/anaconda3/envs/experiment_env/lib/python3.5/site-packages/matplotlib/axes/_axes.py:6499: MatplotlibDeprecationWarning:\n",
      "\n",
      "\n",
      "The 'normed' kwarg was deprecated in Matplotlib 2.1 and will be removed in 3.1. Use 'density' instead.\n",
      "\n",
      "\n",
      "\n",
      "BAM file : ./H3K27ac_H3K27ac/SRX1943216/merged/all-filtered.bam\n",
      "Sample size: 7569\n",
      "\n",
      "Fragment lengths:\n",
      "Min.: 0.0\n",
      "1st Qu.: 110.0\n",
      "Mean: 144.867221561633\n",
      "Median: 146.0\n",
      "3rd Qu.: 184.0\n",
      "Max.: 229.0\n",
      "Std: 48.70437291488072\n",
      "MAD: 37.0\n",
      "Len. 10%: 82.0\n",
      "Len. 20%: 102.0\n",
      "Len. 30%: 118.0\n",
      "Len. 40%: 133.0\n",
      "Len. 60%: 161.0\n",
      "Len. 70%: 176.0\n",
      "Len. 80%: 191.0\n",
      "Len. 90%: 209.0\n",
      "Len. 99%: 227.0\n",
      "\n",
      "\n",
      "Read lengths:\n",
      "Sample size: 7569\n",
      "\n",
      "Min.: 34.0\n",
      "1st Qu.: 42.0\n",
      "Mean: 41.854141894569956\n",
      "Median: 42.0\n",
      "3rd Qu.: 42.0\n",
      "Max.: 42.0\n",
      "Std: 1.0703225011746256\n",
      "MAD: 0.0\n",
      "Len. 10%: 42.0\n",
      "Len. 20%: 42.0\n",
      "Len. 30%: 42.0\n",
      "Len. 40%: 42.0\n",
      "Len. 60%: 42.0\n",
      "Len. 70%: 42.0\n",
      "Len. 80%: 42.0\n",
      "Len. 90%: 42.0\n",
      "Len. 99%: 42.0\n",
      "\n",
      "/homes/dhelekal/anaconda3/envs/experiment_env/lib/python3.5/site-packages/matplotlib/axes/_axes.py:6499: MatplotlibDeprecationWarning:\n",
      "\n",
      "\n",
      "The 'normed' kwarg was deprecated in Matplotlib 2.1 and will be removed in 3.1. Use 'density' instead.\n",
      "\n",
      "\n",
      "\n",
      "BAM file : ./H3K4me3_H3K4me3/SRX1943209/merged/all-filtered.bam\n",
      "Sample size: 15338\n",
      "\n",
      "Fragment lengths:\n",
      "Min.: 0.0\n",
      "1st Qu.: 103.0\n",
      "Mean: 139.61350893206415\n",
      "Median: 140.0\n",
      "3rd Qu.: 178.0\n",
      "Max.: 229.0\n",
      "Std: 48.83839544391686\n",
      "MAD: 37.0\n",
      "Len. 10%: 75.0\n",
      "Len. 20%: 95.0\n",
      "Len. 30%: 111.0\n",
      "Len. 40%: 126.0\n",
      "Len. 60%: 155.0\n",
      "Len. 70%: 169.0\n",
      "Len. 80%: 187.0\n",
      "Len. 90%: 207.0\n",
      "Len. 99%: 227.0\n",
      "\n",
      "\n",
      "Read lengths:\n",
      "Sample size: 15338\n",
      "\n",
      "Min.: 34.0\n",
      "1st Qu.: 42.0\n",
      "Mean: 41.91811187899335\n",
      "Median: 42.0\n",
      "3rd Qu.: 42.0\n",
      "Max.: 42.0\n",
      "Std: 0.8052324532029255\n",
      "MAD: 0.0\n",
      "Len. 10%: 42.0\n",
      "Len. 20%: 42.0\n",
      "Len. 30%: 42.0\n",
      "Len. 40%: 42.0\n",
      "Len. 60%: 42.0\n",
      "Len. 70%: 42.0\n",
      "Len. 80%: 42.0\n",
      "Len. 90%: 42.0\n",
      "Len. 99%: 42.0\n",
      "\n",
      "/homes/dhelekal/anaconda3/envs/experiment_env/lib/python3.5/site-packages/matplotlib/axes/_axes.py:6499: MatplotlibDeprecationWarning:\n",
      "\n",
      "\n",
      "The 'normed' kwarg was deprecated in Matplotlib 2.1 and will be removed in 3.1. Use 'density' instead.\n",
      "\n",
      "\n",
      "\n",
      "BAM file : ./H3K27ac_H3K27ac/SRX1943295/merged/all-filtered.bam\n",
      "Sample size: 8684\n",
      "\n",
      "Fragment lengths:\n",
      "Min.: 0.0\n",
      "1st Qu.: 111.0\n",
      "Mean: 145.70117457392905\n",
      "Median: 148.0\n",
      "3rd Qu.: 186.0\n",
      "Max.: 229.0\n",
      "Std: 49.84087247294637\n",
      "MAD: 38.0\n",
      "Len. 10%: 80.0\n",
      "Len. 20%: 102.0\n",
      "Len. 30%: 119.0\n",
      "Len. 40%: 133.0\n",
      "Len. 60%: 163.0\n",
      "Len. 70%: 178.0\n",
      "Len. 80%: 193.0\n",
      "Len. 90%: 211.0\n",
      "Len. 99%: 228.0\n",
      "\n",
      "\n",
      "Read lengths:\n",
      "Sample size: 8684\n",
      "\n",
      "Min.: 38.0\n",
      "1st Qu.: 38.0\n",
      "Mean: 38.0\n",
      "Median: 38.0\n",
      "3rd Qu.: 38.0\n",
      "Max.: 38.0\n",
      "Std: 0.0\n",
      "MAD: 0.0\n",
      "Len. 10%: 38.0\n",
      "Len. 20%: 38.0\n",
      "Len. 30%: 38.0\n",
      "Len. 40%: 38.0\n",
      "Len. 60%: 38.0\n",
      "Len. 70%: 38.0\n",
      "Len. 80%: 38.0\n",
      "Len. 90%: 38.0\n",
      "Len. 99%: 38.0\n",
      "\n",
      "/homes/dhelekal/anaconda3/envs/experiment_env/lib/python3.5/site-packages/matplotlib/axes/_axes.py:6499: MatplotlibDeprecationWarning:\n",
      "\n",
      "\n",
      "The 'normed' kwarg was deprecated in Matplotlib 2.1 and will be removed in 3.1. Use 'density' instead.\n",
      "\n",
      "\n",
      "\n",
      "BAM file : ./H3K27ac_H3K27ac/SRX1943215/merged/all-filtered.bam\n",
      "Sample size: 7687\n",
      "\n",
      "Fragment lengths:\n",
      "Min.: 0.0\n",
      "1st Qu.: 108.0\n",
      "Mean: 142.63145570443606\n",
      "Median: 145.0\n",
      "3rd Qu.: 180.0\n",
      "Max.: 229.0\n",
      "Std: 48.68979995326366\n",
      "MAD: 36.0\n",
      "Len. 10%: 81.0\n",
      "Len. 20%: 100.0\n",
      "Len. 30%: 115.0\n",
      "Len. 40%: 130.0\n",
      "Len. 60%: 158.0\n",
      "Len. 70%: 173.0\n",
      "Len. 80%: 189.0\n",
      "Len. 90%: 207.0\n",
      "Len. 99%: 227.0\n",
      "\n",
      "\n",
      "Read lengths:\n",
      "Sample size: 7687\n",
      "\n",
      "Min.: 34.0\n",
      "1st Qu.: 42.0\n",
      "Mean: 41.82828151424483\n",
      "Median: 42.0\n",
      "3rd Qu.: 42.0\n",
      "Max.: 42.0\n",
      "Std: 1.1594225492422165\n",
      "MAD: 0.0\n",
      "Len. 10%: 42.0\n",
      "Len. 20%: 42.0\n",
      "Len. 30%: 42.0\n",
      "Len. 40%: 42.0\n",
      "Len. 60%: 42.0\n",
      "Len. 70%: 42.0\n",
      "Len. 80%: 42.0\n",
      "Len. 90%: 42.0\n",
      "Len. 99%: 42.0\n",
      "\n",
      "/homes/dhelekal/anaconda3/envs/experiment_env/lib/python3.5/site-packages/matplotlib/axes/_axes.py:6499: MatplotlibDeprecationWarning:\n",
      "\n",
      "\n",
      "The 'normed' kwarg was deprecated in Matplotlib 2.1 and will be removed in 3.1. Use 'density' instead.\n",
      "\n",
      "\n",
      "\n",
      "BAM file : ./H3K27ac_H3K27ac/SRX1943214/merged/all-filtered.bam\n",
      "Sample size: 7883\n",
      "\n",
      "Fragment lengths:\n",
      "Min.: 0.0\n",
      "1st Qu.: 110.0\n",
      "Mean: 144.76328808829126\n",
      "Median: 147.0\n",
      "3rd Qu.: 183.0\n",
      "Max.: 229.0\n",
      "Std: 48.77152565300239\n",
      "MAD: 37.0\n",
      "Len. 10%: 82.0\n",
      "Len. 20%: 102.0\n",
      "Len. 30%: 117.0\n",
      "Len. 40%: 132.0\n",
      "Len. 60%: 161.0\n",
      "Len. 70%: 176.0\n",
      "Len. 80%: 192.0\n",
      "Len. 90%: 209.0\n",
      "Len. 99%: 227.0\n",
      "\n",
      "\n",
      "Read lengths:\n",
      "Sample size: 7883\n",
      "\n",
      "Min.: 34.0\n",
      "1st Qu.: 42.0\n",
      "Mean: 41.893441583153624\n",
      "Median: 42.0\n",
      "3rd Qu.: 42.0\n",
      "Max.: 42.0\n",
      "Std: 0.917121932226146\n",
      "MAD: 0.0\n",
      "Len. 10%: 42.0\n",
      "Len. 20%: 42.0\n",
      "Len. 30%: 42.0\n",
      "Len. 40%: 42.0\n",
      "Len. 60%: 42.0\n",
      "Len. 70%: 42.0\n",
      "Len. 80%: 42.0\n",
      "Len. 90%: 42.0\n",
      "Len. 99%: 42.0\n",
      "\n",
      "/homes/dhelekal/anaconda3/envs/experiment_env/lib/python3.5/site-packages/matplotlib/axes/_axes.py:6499: MatplotlibDeprecationWarning:\n",
      "\n",
      "\n",
      "The 'normed' kwarg was deprecated in Matplotlib 2.1 and will be removed in 3.1. Use 'density' instead.\n",
      "\n",
      "\n",
      "\n",
      "BAM file : ./H3K27ac_H3K4me3/SRX1943228/merged/all-filtered.bam\n",
      "Sample size: 7225\n",
      "\n",
      "Fragment lengths:\n",
      "Min.: 0.0\n",
      "1st Qu.: 109.0\n",
      "Mean: 144.59944636678202\n",
      "Median: 146.0\n",
      "3rd Qu.: 184.0\n",
      "Max.: 229.0\n",
      "Std: 49.73787734312819\n",
      "MAD: 37.0\n",
      "Len. 10%: 80.0\n",
      "Len. 20%: 100.0\n",
      "Len. 30%: 117.0\n",
      "Len. 40%: 132.0\n",
      "Len. 60%: 161.0\n",
      "Len. 70%: 176.0\n",
      "Len. 80%: 192.0\n",
      "Len. 90%: 211.0\n",
      "Len. 99%: 227.0\n",
      "\n",
      "\n",
      "Read lengths:\n",
      "Sample size: 7225\n",
      "\n",
      "Min.: 34.0\n",
      "1st Qu.: 42.0\n",
      "Mean: 41.83723183391003\n",
      "Median: 42.0\n",
      "3rd Qu.: 42.0\n",
      "Max.: 42.0\n",
      "Std: 1.1294475874636385\n",
      "MAD: 0.0\n",
      "Len. 10%: 42.0\n",
      "Len. 20%: 42.0\n",
      "Len. 30%: 42.0\n",
      "Len. 40%: 42.0\n",
      "Len. 60%: 42.0\n",
      "Len. 70%: 42.0\n",
      "Len. 80%: 42.0\n",
      "Len. 90%: 42.0\n",
      "Len. 99%: 42.0\n",
      "\n",
      "/homes/dhelekal/anaconda3/envs/experiment_env/lib/python3.5/site-packages/matplotlib/axes/_axes.py:6499: MatplotlibDeprecationWarning:\n",
      "\n",
      "\n",
      "The 'normed' kwarg was deprecated in Matplotlib 2.1 and will be removed in 3.1. Use 'density' instead.\n",
      "\n",
      "\n",
      "\n",
      "BAM file : ./H3K27ac_H3K4me3/SRX1943195/merged/all-filtered.bam\n",
      "Sample size: 3437\n",
      "\n",
      "Fragment lengths:\n",
      "Min.: 0.0\n",
      "1st Qu.: 106.0\n",
      "Mean: 143.51585685190574\n",
      "Median: 148.0\n",
      "3rd Qu.: 186.0\n",
      "Max.: 229.0\n",
      "Std: 52.30889094137166\n",
      "MAD: 39.0\n",
      "Len. 10%: 78.0\n",
      "Len. 20%: 97.0\n",
      "Len. 30%: 115.0\n",
      "Len. 40%: 132.0\n",
      "Len. 60%: 163.0\n",
      "Len. 70%: 178.0\n",
      "Len. 80%: 193.0\n",
      "Len. 90%: 211.0\n",
      "Len. 99%: 227.0\n",
      "\n",
      "\n",
      "Read lengths:\n",
      "Sample size: 3437\n",
      "\n",
      "Min.: 34.0\n",
      "1st Qu.: 42.0\n",
      "Mean: 41.81844631946465\n",
      "Median: 42.0\n",
      "3rd Qu.: 42.0\n",
      "Max.: 42.0\n",
      "Std: 1.1914141619801537\n",
      "MAD: 0.0\n",
      "Len. 10%: 42.0\n",
      "Len. 20%: 42.0\n",
      "Len. 30%: 42.0\n",
      "Len. 40%: 42.0\n",
      "Len. 60%: 42.0\n",
      "Len. 70%: 42.0\n",
      "Len. 80%: 42.0\n",
      "Len. 90%: 42.0\n",
      "Len. 99%: 42.0\n",
      "\n",
      "/homes/dhelekal/anaconda3/envs/experiment_env/lib/python3.5/site-packages/matplotlib/axes/_axes.py:6499: MatplotlibDeprecationWarning:\n",
      "\n",
      "\n",
      "The 'normed' kwarg was deprecated in Matplotlib 2.1 and will be removed in 3.1. Use 'density' instead.\n",
      "\n",
      "\n",
      "\n",
      "BAM file : ./H3K27ac_H3K4me3/SRX1943304/merged/all-filtered.bam\n",
      "Sample size: 15491\n",
      "\n",
      "Fragment lengths:\n",
      "Min.: 0.0\n",
      "1st Qu.: 115.0\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mean: 149.77903298689563\n",
      "Median: 155.0\n",
      "3rd Qu.: 189.0\n",
      "Max.: 229.0\n",
      "Std: 50.03804803068047\n",
      "MAD: 37.0\n",
      "Len. 10%: 83.0\n",
      "Len. 20%: 107.0\n",
      "Len. 30%: 124.0\n",
      "Len. 40%: 140.0\n",
      "Len. 60%: 169.0\n",
      "Len. 70%: 183.0\n",
      "Len. 80%: 197.0\n",
      "Len. 90%: 213.0\n",
      "Len. 99%: 228.0\n",
      "\n",
      "\n",
      "Read lengths:\n",
      "Sample size: 15491\n",
      "\n",
      "Min.: 38.0\n",
      "1st Qu.: 38.0\n",
      "Mean: 38.0\n",
      "Median: 38.0\n",
      "3rd Qu.: 38.0\n",
      "Max.: 38.0\n",
      "Std: 0.0\n",
      "MAD: 0.0\n",
      "Len. 10%: 38.0\n",
      "Len. 20%: 38.0\n",
      "Len. 30%: 38.0\n",
      "Len. 40%: 38.0\n",
      "Len. 60%: 38.0\n",
      "Len. 70%: 38.0\n",
      "Len. 80%: 38.0\n",
      "Len. 90%: 38.0\n",
      "Len. 99%: 38.0\n",
      "\n",
      "/homes/dhelekal/anaconda3/envs/experiment_env/lib/python3.5/site-packages/matplotlib/axes/_axes.py:6499: MatplotlibDeprecationWarning:\n",
      "\n",
      "\n",
      "The 'normed' kwarg was deprecated in Matplotlib 2.1 and will be removed in 3.1. Use 'density' instead.\n",
      "\n",
      "\n",
      "\n",
      "BAM file : ./H3K27ac_H3K4me3/SRX1943226/merged/all-filtered.bam\n",
      "Sample size: 5515\n",
      "\n",
      "Fragment lengths:\n",
      "Min.: 0.0\n",
      "1st Qu.: 108.0\n",
      "Mean: 143.0391659111514\n",
      "Median: 145.0\n",
      "3rd Qu.: 184.0\n",
      "Max.: 229.0\n",
      "Std: 50.39914579310946\n",
      "MAD: 38.0\n",
      "Len. 10%: 79.0\n",
      "Len. 20%: 99.0\n",
      "Len. 30%: 116.0\n",
      "Len. 40%: 131.0\n",
      "Len. 60%: 159.0\n",
      "Len. 70%: 175.0\n",
      "Len. 80%: 191.0\n",
      "Len. 90%: 209.0\n",
      "Len. 99%: 227.0\n",
      "\n",
      "\n",
      "Read lengths:\n",
      "Sample size: 5515\n",
      "\n",
      "Min.: 34.0\n",
      "1st Qu.: 42.0\n",
      "Mean: 41.82592928377153\n",
      "Median: 42.0\n",
      "3rd Qu.: 42.0\n",
      "Max.: 42.0\n",
      "Std: 1.1671611352248885\n",
      "MAD: 0.0\n",
      "Len. 10%: 42.0\n",
      "Len. 20%: 42.0\n",
      "Len. 30%: 42.0\n",
      "Len. 40%: 42.0\n",
      "Len. 60%: 42.0\n",
      "Len. 70%: 42.0\n",
      "Len. 80%: 42.0\n",
      "Len. 90%: 42.0\n",
      "Len. 99%: 42.0\n",
      "\n",
      "/homes/dhelekal/anaconda3/envs/experiment_env/lib/python3.5/site-packages/matplotlib/axes/_axes.py:6499: MatplotlibDeprecationWarning:\n",
      "\n",
      "\n",
      "The 'normed' kwarg was deprecated in Matplotlib 2.1 and will be removed in 3.1. Use 'density' instead.\n",
      "\n",
      "\n",
      "\n",
      "BAM file : ./H3K4me3_H3K4me3/SRX1943208/merged/all-filtered.bam\n",
      "Sample size: 11865\n",
      "\n",
      "Fragment lengths:\n",
      "Min.: 0.0\n",
      "1st Qu.: 104.0\n",
      "Mean: 139.7951116729878\n",
      "Median: 141.0\n",
      "3rd Qu.: 178.0\n",
      "Max.: 229.0\n",
      "Std: 49.19283325201849\n",
      "MAD: 37.0\n",
      "Len. 10%: 76.0\n",
      "Len. 20%: 95.0\n",
      "Len. 30%: 112.0\n",
      "Len. 40%: 126.0\n",
      "Len. 60%: 155.0\n",
      "Len. 70%: 170.0\n",
      "Len. 80%: 187.0\n",
      "Len. 90%: 206.0\n",
      "Len. 99%: 227.0\n",
      "\n",
      "\n",
      "Read lengths:\n",
      "Sample size: 11865\n",
      "\n",
      "Min.: 34.0\n",
      "1st Qu.: 42.0\n",
      "Mean: 41.90897597977244\n",
      "Median: 42.0\n",
      "3rd Qu.: 42.0\n",
      "Max.: 42.0\n",
      "Std: 0.8484732108688482\n",
      "MAD: 0.0\n",
      "Len. 10%: 42.0\n",
      "Len. 20%: 42.0\n",
      "Len. 30%: 42.0\n",
      "Len. 40%: 42.0\n",
      "Len. 60%: 42.0\n",
      "Len. 70%: 42.0\n",
      "Len. 80%: 42.0\n",
      "Len. 90%: 42.0\n",
      "Len. 99%: 42.0\n",
      "\n",
      "/homes/dhelekal/anaconda3/envs/experiment_env/lib/python3.5/site-packages/matplotlib/axes/_axes.py:6499: MatplotlibDeprecationWarning:\n",
      "\n",
      "\n",
      "The 'normed' kwarg was deprecated in Matplotlib 2.1 and will be removed in 3.1. Use 'density' instead.\n",
      "\n",
      "\n",
      "\n",
      "BAM file : ./H3K4me3_H3K4me3/SRX1943210/merged/all-filtered.bam\n",
      "Sample size: 14107\n",
      "\n",
      "Fragment lengths:\n",
      "Min.: 0.0\n",
      "1st Qu.: 103.0\n",
      "Mean: 139.90040405472462\n",
      "Median: 140.0\n",
      "3rd Qu.: 178.0\n",
      "Max.: 229.0\n",
      "Std: 48.34024329220287\n",
      "MAD: 37.0\n",
      "Len. 10%: 76.0\n",
      "Len. 20%: 95.20000000000027\n",
      "Len. 30%: 111.0\n",
      "Len. 40%: 125.0\n",
      "Len. 60%: 154.0\n",
      "Len. 70%: 170.0\n",
      "Len. 80%: 187.0\n",
      "Len. 90%: 207.0\n",
      "Len. 99%: 227.0\n",
      "\n",
      "\n",
      "Read lengths:\n",
      "Sample size: 14107\n",
      "\n",
      "Min.: 34.0\n",
      "1st Qu.: 42.0\n",
      "Mean: 41.9194726022542\n",
      "Median: 42.0\n",
      "3rd Qu.: 42.0\n",
      "Max.: 42.0\n",
      "Std: 0.7985828198619661\n",
      "MAD: 0.0\n",
      "Len. 10%: 42.0\n",
      "Len. 20%: 42.0\n",
      "Len. 30%: 42.0\n",
      "Len. 40%: 42.0\n",
      "Len. 60%: 42.0\n",
      "Len. 70%: 42.0\n",
      "Len. 80%: 42.0\n",
      "Len. 90%: 42.0\n",
      "Len. 99%: 42.0\n",
      "\n",
      "/homes/dhelekal/anaconda3/envs/experiment_env/lib/python3.5/site-packages/matplotlib/axes/_axes.py:6499: MatplotlibDeprecationWarning:\n",
      "\n",
      "\n",
      "The 'normed' kwarg was deprecated in Matplotlib 2.1 and will be removed in 3.1. Use 'density' instead.\n",
      "\n",
      "\n",
      "\n",
      "BAM file : ./H3K27ac_H3K4me3/SRX1943303/merged/all-filtered.bam\n",
      "Sample size: 12890\n",
      "\n",
      "Fragment lengths:\n",
      "Min.: 0.0\n",
      "1st Qu.: 115.0\n",
      "Mean: 149.40729247478666\n",
      "Median: 154.0\n",
      "3rd Qu.: 190.0\n",
      "Max.: 229.0\n",
      "Std: 50.68731761639834\n",
      "MAD: 38.0\n",
      "Len. 10%: 83.0\n",
      "Len. 20%: 106.0\n",
      "Len. 30%: 123.0\n",
      "Len. 40%: 139.0\n",
      "Len. 60%: 168.0\n",
      "Len. 70%: 183.0\n",
      "Len. 80%: 198.0\n",
      "Len. 90%: 213.0\n",
      "Len. 99%: 228.0\n",
      "\n",
      "\n",
      "Read lengths:\n",
      "Sample size: 12890\n",
      "\n",
      "Min.: 38.0\n",
      "1st Qu.: 38.0\n",
      "Mean: 38.0\n",
      "Median: 38.0\n",
      "3rd Qu.: 38.0\n",
      "Max.: 38.0\n",
      "Std: 0.0\n",
      "MAD: 0.0\n",
      "Len. 10%: 38.0\n",
      "Len. 20%: 38.0\n",
      "Len. 30%: 38.0\n",
      "Len. 40%: 38.0\n",
      "Len. 60%: 38.0\n",
      "Len. 70%: 38.0\n",
      "Len. 80%: 38.0\n",
      "Len. 90%: 38.0\n",
      "Len. 99%: 38.0\n",
      "\n",
      "/homes/dhelekal/anaconda3/envs/experiment_env/lib/python3.5/site-packages/matplotlib/axes/_axes.py:6499: MatplotlibDeprecationWarning:\n",
      "\n",
      "\n",
      "The 'normed' kwarg was deprecated in Matplotlib 2.1 and will be removed in 3.1. Use 'density' instead.\n",
      "\n",
      "\n",
      "\n",
      "BAM file : ./H3K27ac_H3K4me3/SRX1943197/merged/all-filtered.bam\n",
      "Sample size: 3613\n",
      "\n",
      "Fragment lengths:\n",
      "Min.: 0.0\n",
      "1st Qu.: 108.0\n",
      "Mean: 144.7110434541932\n",
      "Median: 150.0\n",
      "3rd Qu.: 185.0\n",
      "Max.: 229.0\n",
      "Std: 51.52800095281976\n",
      "MAD: 38.0\n",
      "Len. 10%: 78.0\n",
      "Len. 20%: 100.0\n",
      "Len. 30%: 118.0\n",
      "Len. 40%: 134.0\n",
      "Len. 60%: 164.0\n",
      "Len. 70%: 178.0\n",
      "Len. 80%: 193.0\n",
      "Len. 90%: 211.0\n",
      "Len. 99%: 227.0\n",
      "\n",
      "\n",
      "Read lengths:\n",
      "Sample size: 3613\n",
      "\n",
      "Min.: 34.0\n",
      "1st Qu.: 42.0\n",
      "Mean: 41.76086354829781\n",
      "Median: 42.0\n",
      "3rd Qu.: 42.0\n",
      "Max.: 42.0\n",
      "Std: 1.3623161788236902\n",
      "MAD: 0.0\n",
      "Len. 10%: 42.0\n",
      "Len. 20%: 42.0\n",
      "Len. 30%: 42.0\n",
      "Len. 40%: 42.0\n",
      "Len. 60%: 42.0\n",
      "Len. 70%: 42.0\n",
      "Len. 80%: 42.0\n",
      "Len. 90%: 42.0\n",
      "Len. 99%: 42.0\n",
      "\n",
      "/homes/dhelekal/anaconda3/envs/experiment_env/lib/python3.5/site-packages/matplotlib/axes/_axes.py:6499: MatplotlibDeprecationWarning:\n",
      "\n",
      "\n",
      "The 'normed' kwarg was deprecated in Matplotlib 2.1 and will be removed in 3.1. Use 'density' instead.\n",
      "\n",
      "\n",
      "\n",
      "BAM file : ./H3K27ac_H3K4me3/SRX1943198/merged/all-filtered.bam\n",
      "Sample size: 3617\n",
      "\n",
      "Fragment lengths:\n",
      "Min.: 0.0\n",
      "1st Qu.: 108.0\n",
      "Mean: 144.84849322643075\n",
      "Median: 149.0\n",
      "3rd Qu.: 188.0\n",
      "Max.: 229.0\n",
      "Std: 52.03783361949083\n",
      "MAD: 40.0\n",
      "Len. 10%: 77.0\n",
      "Len. 20%: 100.0\n",
      "Len. 30%: 117.0\n",
      "Len. 40%: 133.0\n",
      "Len. 60%: 165.0\n",
      "Len. 70%: 181.0\n",
      "Len. 80%: 193.0\n",
      "Len. 90%: 211.0\n",
      "Len. 99%: 227.0\n",
      "\n",
      "\n",
      "Read lengths:\n",
      "Sample size: 3617\n",
      "\n",
      "Min.: 34.0\n",
      "1st Qu.: 42.0\n",
      "Mean: 41.7367984517556\n",
      "Median: 42.0\n",
      "3rd Qu.: 42.0\n",
      "Max.: 42.0\n",
      "Std: 1.4270029190429017\n",
      "MAD: 0.0\n",
      "Len. 10%: 42.0\n",
      "Len. 20%: 42.0\n",
      "Len. 30%: 42.0\n",
      "Len. 40%: 42.0\n",
      "Len. 60%: 42.0\n",
      "Len. 70%: 42.0\n",
      "Len. 80%: 42.0\n",
      "Len. 90%: 42.0\n",
      "Len. 99%: 42.0\n",
      "\n",
      "/homes/dhelekal/anaconda3/envs/experiment_env/lib/python3.5/site-packages/matplotlib/axes/_axes.py:6499: MatplotlibDeprecationWarning:\n",
      "\n",
      "\n",
      "The 'normed' kwarg was deprecated in Matplotlib 2.1 and will be removed in 3.1. Use 'density' instead.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "find ./*_*/SRX*/merged -mindepth 0 -maxdepth 0 | parallel -j 17 \"bamPEFragmentSize -b {}/all-filtered.bam -o {}/hist.png --numberOfProcessors 2\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tWill parse file: ./H3K27ac_H3K4me3/SRX1943198/merged/all-filtered.bam\n",
      "\n",
      "\tCreating directory: ./H3K27ac_H3K4me3/SRX1943198/tag_dir and removing existing *.tags.tsv\n",
      "\n",
      "\tTreating ./H3K27ac_H3K4me3/SRX1943198/merged/all-filtered.bam as a bam file\n",
      "\tReading alignment file ./H3K27ac_H3K4me3/SRX1943198/merged/all-filtered.bam\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tEstimated genome size = 6321424879\n",
      "\tEstimated average read density = 0.000380 per bp\n",
      "\tTotal Tags = 2399287.0\n",
      "\tTotal Positions = 4084962\n",
      "\tAverage tag length = 38.1\n",
      "\tMedian tags per position = 0 (ideal: 1)\n",
      "\tAverage tags per position = 0.154\n",
      "\tFragment Length Estimate: 155\n",
      "\tPeak Width Estimate: 141\n",
      "\t\t!!! No reliable estimate for peak size\n",
      "\t\tSetting Peak width estimate to be equal to fragment length estimate\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 2.0\n",
      "\t\tDiff strand fold enrichment: 2.2\n",
      "\t\tSame / Diff fold enrichment: 0.9\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq or unstranded RNA-Seq - autocorrelation looks good.\n",
      "\n",
      "\tWill parse file: ./H3K27ac_H3K4me3/SRX1943197/merged/all-filtered.bam\n",
      "\n",
      "\tCreating directory: ./H3K27ac_H3K4me3/SRX1943197/tag_dir and removing existing *.tags.tsv\n",
      "\n",
      "\tTreating ./H3K27ac_H3K4me3/SRX1943197/merged/all-filtered.bam as a bam file\n",
      "\tReading alignment file ./H3K27ac_H3K4me3/SRX1943197/merged/all-filtered.bam\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tEstimated genome size = 6521069760\n",
      "\tEstimated average read density = 0.000380 per bp\n",
      "\tTotal Tags = 2475570.5\n",
      "\tTotal Positions = 4128599\n",
      "\tAverage tag length = 38.1\n",
      "\tMedian tags per position = 0 (ideal: 1)\n",
      "\tAverage tags per position = 0.173\n",
      "\tFragment Length Estimate: 149\n",
      "\tPeak Width Estimate: 149\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 2.0\n",
      "\t\tDiff strand fold enrichment: 2.2\n",
      "\t\tSame / Diff fold enrichment: 0.9\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq or unstranded RNA-Seq - autocorrelation looks good.\n",
      "\n",
      "\tWill parse file: ./H3K27ac_H3K4me3/SRX1943195/merged/all-filtered.bam\n",
      "\n",
      "\tCreating directory: ./H3K27ac_H3K4me3/SRX1943195/tag_dir and removing existing *.tags.tsv\n",
      "\n",
      "\tTreating ./H3K27ac_H3K4me3/SRX1943195/merged/all-filtered.bam as a bam file\n",
      "\tReading alignment file ./H3K27ac_H3K4me3/SRX1943195/merged/all-filtered.bam\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tEstimated genome size = 6413946990\n",
      "\tEstimated average read density = 0.000375 per bp\n",
      "\tTotal Tags = 2403287.0\n",
      "\tTotal Positions = 4027576\n",
      "\tAverage tag length = 38.1\n",
      "\tMedian tags per position = 0 (ideal: 1)\n",
      "\tAverage tags per position = 0.169\n",
      "\tFragment Length Estimate: 152\n",
      "\tPeak Width Estimate: 148\n",
      "\t\t!!! No reliable estimate for peak size\n",
      "\t\tSetting Peak width estimate to be equal to fragment length estimate\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 2.0\n",
      "\t\tDiff strand fold enrichment: 2.2\n",
      "\t\tSame / Diff fold enrichment: 0.9\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq or unstranded RNA-Seq - autocorrelation looks good.\n",
      "\n",
      "\tWill parse file: ./H3K27ac_H3K4me3/SRX1943226/merged/all-filtered.bam\n",
      "\n",
      "\tCreating directory: ./H3K27ac_H3K4me3/SRX1943226/tag_dir and removing existing *.tags.tsv\n",
      "\n",
      "\tTreating ./H3K27ac_H3K4me3/SRX1943226/merged/all-filtered.bam as a bam file\n",
      "\tReading alignment file ./H3K27ac_H3K4me3/SRX1943226/merged/all-filtered.bam\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tEstimated genome size = 6776895483\n",
      "\tEstimated average read density = 0.000571 per bp\n",
      "\tTotal Tags = 3872557.0\n",
      "\tTotal Positions = 6021808\n",
      "\tAverage tag length = 38.1\n",
      "\tMedian tags per position = 0 (ideal: 1)\n",
      "\tAverage tags per position = 0.239\n",
      "\tFragment Length Estimate: 147\n",
      "\tPeak Width Estimate: 148\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 2.0\n",
      "\t\tDiff strand fold enrichment: 2.1\n",
      "\t\tSame / Diff fold enrichment: 1.0\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq or unstranded RNA-Seq - autocorrelation looks good.\n",
      "\n",
      "\tWill parse file: ./H3K27ac_H3K4me3/SRX1943228/merged/all-filtered.bam\n",
      "\n",
      "\tCreating directory: ./H3K27ac_H3K4me3/SRX1943228/tag_dir and removing existing *.tags.tsv\n",
      "\n",
      "\tTreating ./H3K27ac_H3K4me3/SRX1943228/merged/all-filtered.bam as a bam file\n",
      "\tReading alignment file ./H3K27ac_H3K4me3/SRX1943228/merged/all-filtered.bam\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tEstimated genome size = 6364261991\n",
      "\tEstimated average read density = 0.000751 per bp\n",
      "\tTotal Tags = 4780067.0\n",
      "\tTotal Positions = 7449689\n",
      "\tAverage tag length = 38.1\n",
      "\tMedian tags per position = 0 (ideal: 1)\n",
      "\tAverage tags per position = 0.237\n",
      "\tFragment Length Estimate: 147\n",
      "\tPeak Width Estimate: 144\n",
      "\t\t!!! No reliable estimate for peak size\n",
      "\t\tSetting Peak width estimate to be equal to fragment length estimate\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 2.0\n",
      "\t\tDiff strand fold enrichment: 2.1\n",
      "\t\tSame / Diff fold enrichment: 1.0\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq or unstranded RNA-Seq - autocorrelation looks good.\n",
      "\n",
      "\tWill parse file: ./H3K27ac_H3K4me3/SRX1943227/merged/all-filtered.bam\n",
      "\n",
      "\tCreating directory: ./H3K27ac_H3K4me3/SRX1943227/tag_dir and removing existing *.tags.tsv\n",
      "\n",
      "\tTreating ./H3K27ac_H3K4me3/SRX1943227/merged/all-filtered.bam as a bam file\n",
      "\tReading alignment file ./H3K27ac_H3K4me3/SRX1943227/merged/all-filtered.bam\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tEstimated genome size = 7121827933\n",
      "\tEstimated average read density = 0.000718 per bp\n",
      "\tTotal Tags = 5110075.0\n",
      "\tTotal Positions = 7837985\n",
      "\tAverage tag length = 38.1\n",
      "\tMedian tags per position = 0 (ideal: 1)\n",
      "\tAverage tags per position = 0.252\n",
      "\tFragment Length Estimate: 149\n",
      "\tPeak Width Estimate: 142\n",
      "\t\t!!! No reliable estimate for peak size\n",
      "\t\tSetting Peak width estimate to be equal to fragment length estimate\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 1.9\n",
      "\t\tDiff strand fold enrichment: 2.0\n",
      "\t\tSame / Diff fold enrichment: 0.9\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq or unstranded RNA-Seq - autocorrelation looks good.\n",
      "\n",
      "\tWill parse file: ./H3K27ac_H3K27ac/SRX1943215/merged/all-filtered.bam\n",
      "\n",
      "\tCreating directory: ./H3K27ac_H3K27ac/SRX1943215/tag_dir and removing existing *.tags.tsv\n",
      "\n",
      "\tTreating ./H3K27ac_H3K27ac/SRX1943215/merged/all-filtered.bam as a bam file\n",
      "\tReading alignment file ./H3K27ac_H3K27ac/SRX1943215/merged/all-filtered.bam\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tEstimated genome size = 6965862348\n",
      "\tEstimated average read density = 0.000854 per bp\n",
      "\tTotal Tags = 5948561.5\n",
      "\tTotal Positions = 9272227\n",
      "\tAverage tag length = 38.1\n",
      "\tMedian tags per position = 0 (ideal: 1)\n",
      "\tAverage tags per position = 0.238\n",
      "\tFragment Length Estimate: 150\n",
      "\tPeak Width Estimate: 145\n",
      "\t\t!!! No reliable estimate for peak size\n",
      "\t\tSetting Peak width estimate to be equal to fragment length estimate\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 1.5\n",
      "\t\tDiff strand fold enrichment: 1.7\n",
      "\t\tSame / Diff fold enrichment: 0.9\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq or unstranded RNA-Seq - autocorrelation looks good.\n",
      "\n",
      "\tWill parse file: ./H3K27ac_H3K27ac/SRX1943216/merged/all-filtered.bam\n",
      "\n",
      "\tCreating directory: ./H3K27ac_H3K27ac/SRX1943216/tag_dir and removing existing *.tags.tsv\n",
      "\n",
      "\tTreating ./H3K27ac_H3K27ac/SRX1943216/merged/all-filtered.bam as a bam file\n",
      "\tReading alignment file ./H3K27ac_H3K27ac/SRX1943216/merged/all-filtered.bam\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tEstimated genome size = 7071431761\n",
      "\tEstimated average read density = 0.000827 per bp\n",
      "\tTotal Tags = 5846567.5\n",
      "\tTotal Positions = 9333553\n",
      "\tAverage tag length = 38.1\n",
      "\tMedian tags per position = 0 (ideal: 1)\n",
      "\tAverage tags per position = 0.215\n",
      "\tFragment Length Estimate: 154\n",
      "\tPeak Width Estimate: 142\n",
      "\t\t!!! No reliable estimate for peak size\n",
      "\t\tSetting Peak width estimate to be equal to fragment length estimate\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 1.5\n",
      "\t\tDiff strand fold enrichment: 1.7\n",
      "\t\tSame / Diff fold enrichment: 0.9\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq or unstranded RNA-Seq - autocorrelation looks good.\n",
      "\n",
      "\tWill parse file: ./H3K27ac_H3K27ac/SRX1943214/merged/all-filtered.bam\n",
      "\n",
      "\tCreating directory: ./H3K27ac_H3K27ac/SRX1943214/tag_dir and removing existing *.tags.tsv\n",
      "\n",
      "\tTreating ./H3K27ac_H3K27ac/SRX1943214/merged/all-filtered.bam as a bam file\n",
      "\tReading alignment file ./H3K27ac_H3K27ac/SRX1943214/merged/all-filtered.bam\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tEstimated genome size = 6927735055\n",
      "\tEstimated average read density = 0.000887 per bp\n",
      "\tTotal Tags = 6141920.0\n",
      "\tTotal Positions = 9592677\n",
      "\tAverage tag length = 38.1\n",
      "\tMedian tags per position = 0 (ideal: 1)\n",
      "\tAverage tags per position = 0.236\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tFragment Length Estimate: 149\n",
      "\tPeak Width Estimate: 147\n",
      "\t\t!!! No reliable estimate for peak size\n",
      "\t\tSetting Peak width estimate to be equal to fragment length estimate\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 1.6\n",
      "\t\tDiff strand fold enrichment: 1.7\n",
      "\t\tSame / Diff fold enrichment: 0.9\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq or unstranded RNA-Seq - autocorrelation looks good.\n",
      "\n",
      "\tWill parse file: ./H3K27ac_H3K27ac/SRX1943295/merged/all-filtered.bam\n",
      "\n",
      "\tCreating directory: ./H3K27ac_H3K27ac/SRX1943295/tag_dir and removing existing *.tags.tsv\n",
      "\n",
      "\tTreating ./H3K27ac_H3K27ac/SRX1943295/merged/all-filtered.bam as a bam file\n",
      "\tReading alignment file ./H3K27ac_H3K27ac/SRX1943295/merged/all-filtered.bam\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tEstimated genome size = 7264206149\n",
      "\tEstimated average read density = 0.000977 per bp\n",
      "\tTotal Tags = 7096986.5\n",
      "\tTotal Positions = 12197120\n",
      "\tAverage tag length = 38.0\n",
      "\tMedian tags per position = 0 (ideal: 1)\n",
      "\tAverage tags per position = 0.144\n",
      "\tFragment Length Estimate: 157\n",
      "\tPeak Width Estimate: 150\n",
      "\t\t!!! No reliable estimate for peak size\n",
      "\t\tSetting Peak width estimate to be equal to fragment length estimate\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 1.5\n",
      "\t\tDiff strand fold enrichment: 1.6\n",
      "\t\tSame / Diff fold enrichment: 0.9\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq or unstranded RNA-Seq - autocorrelation looks good.\n",
      "\n",
      "\tWill parse file: ./H3K27ac_H3K4me3/SRX1943303/merged/all-filtered.bam\n",
      "\n",
      "\tCreating directory: ./H3K27ac_H3K4me3/SRX1943303/tag_dir and removing existing *.tags.tsv\n",
      "\n",
      "\tTreating ./H3K27ac_H3K4me3/SRX1943303/merged/all-filtered.bam as a bam file\n",
      "\tReading alignment file ./H3K27ac_H3K4me3/SRX1943303/merged/all-filtered.bam\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tEstimated genome size = 7462816076\n",
      "\tEstimated average read density = 0.001193 per bp\n",
      "\tTotal Tags = 8900789.0\n",
      "\tTotal Positions = 9968746\n",
      "\tAverage tag length = 38.0\n",
      "\tMedian tags per position = 0 (ideal: 1)\n",
      "\tAverage tags per position = 0.556\n",
      "\tFragment Length Estimate: 158\n",
      "\tPeak Width Estimate: 160\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 2.3\n",
      "\t\tDiff strand fold enrichment: 2.3\n",
      "\t\tSame / Diff fold enrichment: 1.0\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq or unstranded RNA-Seq - autocorrelation looks good.\n",
      "\n",
      "\tWill parse file: ./H3K27ac_H3K27ac/SRX1943296/merged/all-filtered.bam\n",
      "\n",
      "\tCreating directory: ./H3K27ac_H3K27ac/SRX1943296/tag_dir and removing existing *.tags.tsv\n",
      "\n",
      "\tTreating ./H3K27ac_H3K27ac/SRX1943296/merged/all-filtered.bam as a bam file\n",
      "\tReading alignment file ./H3K27ac_H3K27ac/SRX1943296/merged/all-filtered.bam\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tEstimated genome size = 7464812624\n",
      "\tEstimated average read density = 0.001137 per bp\n",
      "\tTotal Tags = 8485154.0\n",
      "\tTotal Positions = 14519860\n",
      "\tAverage tag length = 38.0\n",
      "\tMedian tags per position = 0 (ideal: 1)\n",
      "\tAverage tags per position = 0.147\n",
      "\tFragment Length Estimate: 155\n",
      "\tPeak Width Estimate: 154\n",
      "\t\t!!! No reliable estimate for peak size\n",
      "\t\tSetting Peak width estimate to be equal to fragment length estimate\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 1.5\n",
      "\t\tDiff strand fold enrichment: 1.6\n",
      "\t\tSame / Diff fold enrichment: 0.9\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq or unstranded RNA-Seq - autocorrelation looks good.\n",
      "\n",
      "\tWill parse file: ./H3K4me3_H3K4me3/SRX1943208/merged/all-filtered.bam\n",
      "\n",
      "\tCreating directory: ./H3K4me3_H3K4me3/SRX1943208/tag_dir and removing existing *.tags.tsv\n",
      "\n",
      "\tTreating ./H3K4me3_H3K4me3/SRX1943208/merged/all-filtered.bam as a bam file\n",
      "\tReading alignment file ./H3K4me3_H3K4me3/SRX1943208/merged/all-filtered.bam\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tEstimated genome size = 7400485397\n",
      "\tEstimated average read density = 0.001157 per bp\n",
      "\tTotal Tags = 8564989.0\n",
      "\tTotal Positions = 14034424\n",
      "\tAverage tag length = 38.1\n",
      "\tMedian tags per position = 0 (ideal: 1)\n",
      "\tAverage tags per position = 0.189\n",
      "\tFragment Length Estimate: 136\n",
      "\tPeak Width Estimate: 145\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 1.8\n",
      "\t\tDiff strand fold enrichment: 1.9\n",
      "\t\tSame / Diff fold enrichment: 1.0\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq or unstranded RNA-Seq - autocorrelation looks good.\n",
      "\n",
      "\tWill parse file: ./H3K27ac_H3K4me3/SRX1943304/merged/all-filtered.bam\n",
      "\n",
      "\tCreating directory: ./H3K27ac_H3K4me3/SRX1943304/tag_dir and removing existing *.tags.tsv\n",
      "\n",
      "\tTreating ./H3K27ac_H3K4me3/SRX1943304/merged/all-filtered.bam as a bam file\n",
      "\tReading alignment file ./H3K27ac_H3K4me3/SRX1943304/merged/all-filtered.bam\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tEstimated genome size = 7579322652\n",
      "\tEstimated average read density = 0.001433 per bp\n",
      "\tTotal Tags = 10862266.0\n",
      "\tTotal Positions = 11899338\n",
      "\tAverage tag length = 38.0\n",
      "\tMedian tags per position = 0 (ideal: 1)\n",
      "\tAverage tags per position = 0.578\n",
      "\tFragment Length Estimate: 157\n",
      "\tPeak Width Estimate: 164\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 2.2\n",
      "\t\tDiff strand fold enrichment: 2.3\n",
      "\t\tSame / Diff fold enrichment: 1.0\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq or unstranded RNA-Seq - autocorrelation looks good.\n",
      "\n",
      "\tWill parse file: ./H3K4me3_H3K4me3/SRX1943210/merged/all-filtered.bam\n",
      "\n",
      "\tCreating directory: ./H3K4me3_H3K4me3/SRX1943210/tag_dir and removing existing *.tags.tsv\n",
      "\n",
      "\tTreating ./H3K4me3_H3K4me3/SRX1943210/merged/all-filtered.bam as a bam file\n",
      "\tReading alignment file ./H3K4me3_H3K4me3/SRX1943210/merged/all-filtered.bam\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tEstimated genome size = 7316455007\n",
      "\tEstimated average read density = 0.001347 per bp\n",
      "\tTotal Tags = 9858786.0\n",
      "\tTotal Positions = 16301300\n",
      "\tAverage tag length = 38.1\n",
      "\tMedian tags per position = 0 (ideal: 1)\n",
      "\tAverage tags per position = 0.180\n",
      "\tFragment Length Estimate: 145\n",
      "\tPeak Width Estimate: 133\n",
      "\t\t!!! No reliable estimate for peak size\n",
      "\t\tSetting Peak width estimate to be equal to fragment length estimate\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 1.8\n",
      "\t\tDiff strand fold enrichment: 1.8\n",
      "\t\tSame / Diff fold enrichment: 1.0\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq or unstranded RNA-Seq - autocorrelation looks good.\n",
      "\n",
      "\tWill parse file: ./H3K4me3_H3K4me3/SRX1943209/merged/all-filtered.bam\n",
      "\n",
      "\tCreating directory: ./H3K4me3_H3K4me3/SRX1943209/tag_dir and removing existing *.tags.tsv\n",
      "\n",
      "\tTreating ./H3K4me3_H3K4me3/SRX1943209/merged/all-filtered.bam as a bam file\n",
      "\tReading alignment file ./H3K4me3_H3K4me3/SRX1943209/merged/all-filtered.bam\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tEstimated genome size = 7508705130\n",
      "\tEstimated average read density = 0.001418 per bp\n",
      "\tTotal Tags = 10648203.0\n",
      "\tTotal Positions = 17290150\n",
      "\tAverage tag length = 38.1\n",
      "\tMedian tags per position = 0 (ideal: 1)\n",
      "\tAverage tags per position = 0.197\n",
      "\tFragment Length Estimate: 139\n",
      "\tPeak Width Estimate: 141\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 1.8\n",
      "\t\tDiff strand fold enrichment: 1.8\n",
      "\t\tSame / Diff fold enrichment: 1.0\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq or unstranded RNA-Seq - autocorrelation looks good.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "find ./*_*/SRX* -mindepth 0 -maxdepth 0 | parallel -j 40 \"makeTagDirectory {}/tag_dir {}/merged/all-filtered.bam\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "find ./*_*/SRX* -mindepth 0 -maxdepth 0 | parallel -j 40 \"findPeaks {}/tag_dir -style histone -size 200 -minDist 100 -o auto\""
   ]
  }
 ],
 "metadata": {
  "celltoolbar": "Raw Cell Format",
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
